### DISCUSSION TITLE
Quick Input Service OnType Functionality

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#10017] [General]
Hi, 
I'm trying to access the oninput functionality in Quick Input. i.e. the function should run every time the value stored in the Quick Picker is changed. Is this functionality available in QuickInput ? Else is there any way to implement this ?

**Comment by msujew** [comment]
Hi @maheshrajrp, do you mean when the user switches between different proposed quick input items? Or rather when the user types something into the search bar?

For the former, you can take a look at the theme selection quick input:

https://github.com/eclipse-theia/theia/blob/1f1aecb5ae3241c096c5d2a59a6c3e436de87171/packages/core/src/browser/common-frontend-contribution.ts#L1099-L1117

**Reply by maheshrajrp** [reply]
I meant when the user types into the field.  Similar to onChange in Form Input.

**Reply by msujew** [reply]
Then `onDidChangeValue` is the function you're looking for:

https://github.com/eclipse-theia/theia/blob/1f1aecb5ae3241c096c5d2a59a6c3e436de87171/packages/git/src/browser/git-quick-open-service.ts#L70-L74
